lib/serviceDiscovery/engines/consul/index.js

```diff
@@ -88,20 +89,52 @@ class ConsulService extends Service {
 	announce(port, metadata, cb) {
 		logger.debug(`Announcing service ${this.name} on port ${port}`);
 		this._announces.push({port, metadata});
-		this.session((err) => {
+		this.session((err, id) => {
 			if (err) {
 				return cb(err);
 			}
 
 			const key = [this.path, helpers.generateId(port)].join('/');
-			this._consul.kv.set({
-				key,
-				value: JSON.stringify({
-					port,
-					ip: this._ip,
-					data: metadata
-				})
-			}, cb);
+
+			const doSet = () => {
+				this._consul.kv.set({
+					key,
+					value: JSON.stringify({
+						port,
+						ip: this._ip,
+						data: metadata
+					}),
+					acquire: id
+				}, (err, result) => {
+					if (err) {
+						return cb(err);
+					}
+
+					// If we can contact Consul but we can't set the key, there is no 'err'; the
+					// returned result is simply set to false so we need to check for it here
+					if (result === false) {
+						return cb(new Error('Could not set Consul KV for service (likely due to another lock)'));
```

Comment:
Could you look into using MageError, which would provide a better logging facility built-in?

Reply:
Gotcha, I'll change it.

Reply:
After looking into this, I'm not sure that's the correct way to go.  MageError seems geared towards sending clients error information, but this is a critical error that will prevent the application from even starting up as it's an uncaught exception down the pipe and regular Error is being used elsewhere in the file.  Thoughts?

Reply:
IIRC it will help you provide more contextual data once the error gets logged. So I believe it should provide additional value, but you might want to test to see how that would look.

Reply:
It does give the description field which I can put some stuff in.  The code/level is wasted but it's at least a bit of extra info, yeah.  I'll use it.  Pushing shortly.